Rep 4029 - Add stripped links to xml response

JoelRizner opened the pull request.

wdschei commented on repose-aggregator/components/filters/uri-stripper-filter/src/main/resources/META-INF/schema/examples/uri-stripper.cfg.xml:
```diff
@@ -3,8 +3,19 @@
 <uri-stripper xmlns="http://docs.openrepose.org/repose/uri-stripper/v1.0" rewrite-location="true" token-index="1">
     <link-resource uri-path-regex=".*" http-methods="ALL">
         <response>
+
```
Why the extra white space?

wdschei commented on repose-aggregator/components/filters/uri-stripper-filter/src/main/scala/org/openrepose/filters/uristripper/UriStripperFilter.scala:
```diff
@@ -171,7 +192,7 @@ class UriStripperFilter @Inject()(configurationService: ConfigurationService)
     Option(contentType) match {
       case Some(ct) if ct.toLowerCase.contains("json") => resource.getJson.toList
       // todo: return xpath when xml is supported
```
Should this `TODO` be removed now?

wdschei commented on repose-aggregator/components/filters/uri-stripper-filter/src/main/scala/org/openrepose/filters/uristripper/UriStripperFilter.scala:
```diff
@@ -231,6 +252,20 @@ class UriStripperFilter @Inject()(configurationService: ConfigurationService)
+
+
+
+
```
Why all the extra white space?

JoelRizner replied:
yeah.. that was where i finished friday so i pushed, in retrospect i should have just stashed the unfinished changes.

wdschei commented:
The following test seems to be missing from the `UriStripperSchemaTest.scala`
```scala
it(s"should successfully validate if only json is present") {
  val config =
  s"""<uri-stripper xmlns="http://docs.openrepose.org/repose/uri-stripper/v1.0" rewrite-location="true" token-index="1">
            |    <link-resource>
            |        <request>
            |            <json>$$.service.link</json>
            |        </request>
            |        <response>
            |            <json>$$.service.link</json>
            |        </response>
            |    </link-resource>
            |</uri-stripper>""".stripMargin

  validator.validateConfigString(config)
}
```

adrianjgeorge commented:
retest this please

JoelRizner commented:
Could this failure be because i added saxon HE in to the properties? The failure is in the saxonEE functionality tests in the translation filter. I added saxon HE because Jorge was saying EE needed a license so i wasn't sure there..

wdschei commented:
That is possible. I was wondering why you needed to add that.

wdschei commented:
The EE only needs a license if you do EE stuff. If you are not doing EE stuff then it should be good.

adrianjgeorge commented:
Yeah, the ee tests are losing the classpath war. he is  showing up first

wdschei commented on repose-aggregator/components/filters/uri-stripper-filter/build.gradle:
```diff
@@ -6,6 +6,9 @@ dependencies {
     compile "com.typesafe.play:play-json_$scalaMajDotMin"
     compile "io.gatling:jsonpath_$scalaMajDotMin"
+    compile "org.scala-lang.modules:scala-xml_$scalaMajDotMin"
+    compile "com.rackspace.cloud.api:wadl-tools_$scalaMajDotMin"
```
Is Repose actually using WADL Tools now?

wdschei commented on repose-aggregator/components/filters/uri-stripper-filter/src/main/scala/org/openrepose/filters/uristripper/UriStripperFilter.scala:
```diff
@@ -127,21 +124,81 @@ class UriStripperFilter @Inject()(configurationService: ConfigurationService)
+        val out = new ByteArrayOutputStream()
+        val transformer = templateMap.get(linkPath).get.newTransformer
+        transformer.asInstanceOf[Controller].addLogErrorListener
+        transformer.setParameter("removedToken", token.getOrElse("")) //todo make optional
```
What is the plan for making this an Optional?

JoelRizner replied:
I can remove that todo, it was from a while back. We agreed to pass an empty string instead and handle that logic in the xslt

JoelRizner commented:
We are using WADL tools for the transformation, provided by Jorge

wdschei commented:
This looks good. Way to knock out the Marshaller!!!
